pkgs/development/interpreters/python/mk-python-derivation.nix

```
@@ -314,7 +315,7 @@ let
           unzip
         ]
         ++ optionals (format' == "setuptools") [
-          setuptoolsBuildHook
+          (lib.warn "${name} uses the deprecated setuptools format. Remove `format` and configure `pyproject = true` and `build-system` instead." setuptoolsBuildHook)
```

```suggestion
          (lib.warn "${name} uses the deprecated setuptools format. Remove `format` and configure `pyproject = true` and `build-system = [ setuptools ]` instead." setuptoolsBuildHook)
```
in case you want to avoid people having to look into the manual